applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1436,87 +1362,90 @@ int unicast_client_discover(struct bt_conn *conn, enum unicast_discover_dir dir)
 	return ret;
 }
 
-int unicast_client_start(enum bt_audio_dir dir)
+int unicast_client_start(void)
 {
 	int ret;
-	bool any_errors = false;
+	struct bt_cap_unicast_audio_start_stream_param
+		cap_stream_params[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT +
+				  CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
+	static struct bt_cap_unicast_audio_start_param param;
+
+	param.stream_params = cap_stream_params;
+	param.count = 0;
+	param.type = BT_CAP_SET_TYPE_AD_HOC;
 
 	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
-		if (dir == BT_AUDIO_DIR_SINK &&
-		    le_audio_ep_state_check(headsets[i].sink_stream.ep,
-					    BT_BAP_EP_STATE_QOS_CONFIGURED)) {
+		if (le_audio_ep_state_check(headsets[i].sink_ep, BT_BAP_EP_STATE_IDLE)) {
```

Comment:
Should we have some check if the endpoint is not in IDLE state?

Reply:
If it's not in IDLE state we can't start it, at least not until host has been changed

Reply:
Added a helper function for getting the ep state and included that in the WRN print if any are found not to be in IDLE state